resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/HadoopUGIUtil.scala

```diff
@@ -67,6 +67,7 @@ private[spark] class HadoopUGIUtil{
     val byteStream = new ByteArrayOutputStream
     val dataStream = new DataOutputStream(byteStream)
     creds.writeTokenStorageToStream(dataStream)
+    dataStream.close()
```

Comment:
We need to make sure this is called even if `creds.writeTokenStorageToStream(dataStream)` throws an exception (unlikely but still worth considering). Not sure what's the best practice to do this in Scala.

Reply:
+1